src/main.cpp

```diff
@@ -543,3 +550,25 @@ void exit_handler(int s)
         exit( exit_status );
     }
 }
+
+void hangup_handler(int signum)
```

Comment:
The content of this function looks suspiciously similar to that of `exit_handler`. Is there a need to have a copy of it here, or can it be extracted into a separate function, called from both signal handlers?

----

The parameter name is inconsistent: in the declaration at the top of the file it's called `dummy`, here it's called `signum`, which one is it? If the value of a parameter is not used, just leave out its name (but keep the type):
```C++
void handler(int);
```
This allows to get rid of the `(void) signum;` and shows the intent clearly.